command/snapshot/inspect/snapshot_inspect.go

```diff
@@ -145,19 +186,48 @@ type typeStats struct {
 	Count int
 }
 
-func generatetypeStats(info map[structs.MessageType]typeStats) []typeStats {
-	ss := make([]typeStats, 0, len(info))
+func generateStats(info SnapshotInfo) []typeStats {
+	ss := make([]typeStats, 0, len(info.Stats))
 
-	for _, s := range info {
+	for _, s := range info.Stats {
 		ss = append(ss, s)
 	}
 
-	// Sort the stat slice
-	sort.Slice(ss, func(i, j int) bool { return ss[i].Sum > ss[j].Sum })
+	ss = sortTypeStats(ss)
 
 	return ss
 }
 
+func generateKVStats(info SnapshotInfo) []typeStats {
```

Comment:
I don't know if you _have_ to do it, I know Go wants it for exported funcs, but a summary about the func itself is my personal preference. You do a great job commenting throughout as well.